solidity/test/cross-chain/wormhole/L1BTCDepositorNtt.core.test.ts

```diff
@@ -0,0 +1,613 @@
+                await createSnapshot()
+
+                await l1BtcDepositorNtt
+                  .connect(relayer)
+                  .initializeDeposit(
+                    initializeDepositFixture.fundingTx,
+                    initializeDepositFixture.reveal,
+                    initializeDepositFixture.destinationChainDepositOwner
+                  )
+              })
+
+              after(async () => {
+                bridge.revealDepositWithExtraData.reset()
+                await restoreSnapshot()
+              })
+
+              it("should revert", async () => {
+                await expect(
+                  l1BtcDepositorNtt
+                    .connect(relayer)
+                    .initializeDeposit(
+                      initializeDepositFixture.fundingTx,
+                      initializeDepositFixture.reveal,
+                      initializeDepositFixture.destinationChainDepositOwner
+                    )
+                ).to.be.revertedWith("Wrong deposit state")
+              })
+            })
+          })
+
+          context("when the deposit state is correct", () => {
+            before(async () => {
+              await createSnapshot()
+              bridge.revealDepositWithExtraData.returns(
+                initializeDepositFixture.depositKey
+              )
+            })
+
+            after(async () => {
+              bridge.revealDepositWithExtraData.reset()
+              await restoreSnapshot()
+            })
+
+            it("should initialize deposit successfully", async () => {
+              const tx = await l1BtcDepositorNtt
+                .connect(relayer)
+                .initializeDeposit(
+                  initializeDepositFixture.fundingTx,
+                  initializeDepositFixture.reveal,
+                  initializeDepositFixture.destinationChainDepositOwner
+                )
+
+              await expect(tx)
+                .to.emit(l1BtcDepositorNtt, "DepositInitialized")
+                .withArgs(
+                  initializeDepositFixture.depositKey,
+                  initializeDepositFixture.destinationChainDepositOwner.toLowerCase(),
+                  relayer.address
+                )
+
+              expect(
+                await l1BtcDepositorNtt.deposits(
+                  initializeDepositFixture.depositKey
+                )
+              ).to.equal(1)
+            })
+          })
+        })
+      })
+    })
+
+    describe("finalizeDeposit", () => {
+      context("when the deposit state is wrong", () => {
+        context("when the deposit state is Unknown", () => {
+          it("should revert", async () => {
+            await expect(
+              l1BtcDepositorNtt
+                .connect(relayer)
+                .finalizeDeposit(initializeDepositFixture.depositKey)
+            ).to.be.revertedWith("Wrong deposit state")
+          })
+        })
+      })
+
+      context("when the deposit state is correct", () => {
```

Comment:
There isn't actually any `it()` clause in that test, so we aren't assertiong anything here

Reply:
there is an `it()` down below, maybe is confusing with the other `before()` and `after()` over there:

```javascript
it("should initialize deposit successfully", async () => {
```